scripts/src/public/services/offers.ts

```diff
@@ -72,12 +72,10 @@ async function filterOffers(userId: string, app: Application | undefined, logger
 	const offerCounts = await Order.countAllByOffer(userId);
 	const contents = await offerContents.getAllContents();
 	let availableTranslations: OfferTranslation[] = [];
-	let language: string | null = null;
+	let language: string | false = false;
 	if (acceptsLanguagesFunc) {
-		availableTranslations = await OfferTranslation.createQueryBuilder("translations")
-			.where("translations.language IN (:languages)", { languages: acceptsLanguagesFunc() })
-			.getMany();
-		const availableLanguages = new Set(availableTranslations.map(translation => translation.language));
+		let availableLanguages;
+		[availableLanguages, availableTranslations] = await OfferTranslation.getSupportedLanguages({ languages: acceptsLanguagesFunc() });
```

Comment:
The api for `getSupportedLanguages` doesn't feel clean. Why would it return the `availableLanguages` if you can deduce it from the `availableTranslations`

Reply:
Helper function. We do this in a few places, DRY.